src/vehicle.cpp

```diff
@@ -4312,6 +4312,21 @@ void vehicle::slow_leak()
     }
 }
 
+int vehicle::slowdown( const bool should_fall ) const
+{
+    int relative_dir = std::abs( face.dir() - move.dir() ) % 180;
+    // Mph lost per tile when coasting, by an ideal vehicle
+    int slowdown = should_fall ? 1 : 5 + std::floor( 45 * sin( DEGREES( relative_dir ) ) );
+
+    // "Anti-ideal" vehicle slows down up to 10 times faster than ideal one
+    const float k_slowdown = 20.0f / ( 2.0f + 9 * ( k_dynamics() * k_mass() ) );
+    // drag is in units of 1/2 HP here, so plows make good emergency brakes.
+    slowdown = drag() + ( int )ceil( k_slowdown * slowdown );
```

Comment:
Using same variable for different purposes is error-prone and makes reading harder than it needs to be. Also, the cast is C-ish. Should be `static_cast<int>( std::ceil( ... ) )`.

Reply:
Fixed